tests/Unit/CssInlinerTest.php

```diff
@@ -3206,6 +3206,121 @@ public function inlineCssNotCopiesInlinableRuleAfterImportRuleToStyleElement()
         self::assertNotContainsCss($cssAfter, $subject->render());
     }
 
+    /**
+     * @return string[][]
+     */
+    public function provideValidFontFaceRules(): array
+    {
+        return [
+            'single @font-face' => [
+                'before' => '',
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => '',
+            ],
+            'uppercase @FONT-FACE' => [
+                'before' => '',
+                '@font-face' => '@FONT-FACE { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => '',
+            ],
+            'mixed case @FoNt-FaCe' => [
+                'before' => '',
+                '@font-face' => '@FoNt-FaCe { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => '',
+            ],
+            '2 @font-faces' => [
+                'before' => '',
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }'
+                    . "\n" . '@font-face { font-family: "Bar Sans"; src: url("/bar-sans.woff2") format("woff2"); }',
+                'after' => '',
+            ],
+            '2 @font-faces, minified' => [
+                'before' => '',
+                '@font-face' => '@font-face{font-family:"Foo Sans";src:url(/foo-sans.woff2) format("woff2")}'
+                    . '@font-face{font-family:"Bar Sans";src:url(/bar-sans.woff2) format("woff2")}',
+                'after' => '',
+            ],
+            '@font-face followed by matching inlinable rule' => [
+                'before' => '',
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => "\n" . 'p { color: green; }',
+            ],
+            '@font-face followed by matching uninlinable rule' => [
+                'before' => '',
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => "\n" . 'p:hover { color: green; }',
+            ],
+            '@font-face followed by matching @media rule' => [
+                'before' => '',
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => "\n" . '@media (max-width: 640px) { p { color: green; } }',
+            ],
+            '@font-face preceded by matching inlinable rule' => [
+                'before' => 'p { color: green; }' . "\n",
+                '@font-face' => '@font-face { font-family: "Foo Sans"; src: url("/foo-sans.woff2") format("woff2"); }',
+                'after' => '',
+            ],
+            '@font-face preceded by matching uninlinable rule' => [
+                'before' => 'p:hover { color: green; }' . "\n",
```

Comment:
As there are not double quotes in the string, we can use a single string encased in double quotes here, allowing us to avoid the concatenation altogether here. Ditto for the next data set.

Reply:
Done.